docs/process/testing.md

```diff
@@ -0,0 +1,366 @@
+  - [ ] Avatar of user is shown
+  - [ ] User can 'at-mention' those associated with the respective repo; either subject or description field is ok (published repositories only)
+  - [ ] User can 'pound-mention' an issue in the either subject or description field; issue number should populate (published repositories only)
+  - [ ] Description field is optional
+  - [ ] User can undo last commit
+    - [ ] `Push` with number of commits badge is decremented or reverts to `Fetch origin`
+  - [ ] `Undo` button disabled if user is pushing commit
+  - [ ] User can publish a new repo with no commits (aka unborn repo/branch)
+
+### Branches list
+  - [ ] Current branch always shows if repository present
+    - [ ] Hover shows full branch name in tooltip
+  - [ ] Opening list shows all branches in categorized format with a working filter
+    - [ ] `New` button opens 'New Branch' modal
+    - [ ] If filters results are nil, then prefill branch name in modal if user clicks `Create New Branch`
+    - [ ] Active branch is highlighted and marked with a check
+    - [ ] `ESC` clears the filter
+    - [ ] Hover shows full branch name in tooltip
+  - [ ] Default branch labeled and listed first, with timestamp
+  - [ ] Selecting a branch switches branches
+  - [ ] Creating a new branch shows `Publish branch` button
+    - [ ] Publishing successful if logged in only; else error message
+      - [ ] `Create Pull Request` menu option shows warning if branch not published yet
+  - [ ] Renamed branches updated on .com and vice-versa if logged in; else error message
+    - [ ] Opens modal with ability to enter new name
+  - [ ] Deleted branches updated on .com and vice-versa if logged in; else error message
+    - [ ] Deleting branch show warning message
+
+### Fetching origin/Pull
+  - [ ] Code is constantly being fetched from .com with timestamp
+     - [ ] Hover shows timestamp in tooltip
+    - [ ] If Pull Requests on .com, they are reflected with down arrow and quantity
+    - [ ] Pull Requests and Commits can co-exist; error surfaces if merge commit
+  - [ ] User cannot Push/Pull without being signed in; error message surfaced
+    - [ ] Push/Pull works with public/private/Enterprise repos
+    - [ ] Tooltip shows status upon hover, if progress to display
+  - [ ] When a branch is local-only make sure that the `Fetch` button changes to `Publish` and it publishes
+  
+### Publishing only
+  - [ ] Unpublished repository, unborn HEAD - `Publish button` enabled (user can publish repository)
+  - [ ] Unpublished repository, valid branch - `Publish button` enabled (user can publish repository and branch)
+  - [ ] Published repository, unborn HEAD - `Publish button` disabled (no branch to push)
+  - [ ] Published repository, branch without tracking - `Publish button` enabled (user can publish branch)
+  - [ ] Published repository, network action - `Publish button` disabled (don't interfere with existing action)
+  
+### Dotcom
```

Comment:
Nit: "Dotcom" is [slang](https://en.wikipedia.org/wiki/Slang) - maybe we could say "GitHub The Website" here for those who aren't up on the lingo?